path_tracing/matrix_shortest_path_using_depth_first_search.py

```diff
@@ -0,0 +1,108 @@
+"""
+This is an implementation for finding the shortest path of a
+non symmetrical matrix between given start and end points including
+consideration of obstacles using depth first search approach
+
+For doctests run following command:
+python3 -m doctest -v matrix_shortest_path_using_depth_first_search.py
+
+For manual testing run:
+python3 matrix_shortest_path_using_depth_first_search.py
+"""
+
+import math
+
+OBSTACLE_POS = -1
+VALID_POS = 1
+START_POS = "s"
+END_POS = "e"
+VISITED_POS = None
+current_shortest_path_distance = math.inf
+
+
+def find_shortest_path_using_dfs(
+    matrix: list, row: int, col: int, distance: int
+) -> None:
+    """
+    Examples:
+    >>> find_shortest_path_using_dfs([['s', 1, 1], [1, -1, 1], [-1, 1, 'e']], 0, 0, 0)
+    None
+    >>> find_shortest_path_using_dfs([[-1, 1], ['s', 1], [-1, 1], [1, 'e']], 1, 0, 0)
+    None
+    >>> find_shortest_path_using_dfs([[1, 1, 1], [1, 1,'s'], [1, -1, 'e']], 1, 2, 0)
+    None
+    >>> find_shortest_path_using)dfs([[1], [-1, 's', 1], [-1, -1, 'e', -1]], 2, 1, 0)
```

Comment:
There is no point of these doctests if they return `None`. All this doctest is really doing is checking for errors, which only needs to be done once.